core/treepath/treepath.go

```diff
@@ -0,0 +1,257 @@
+// Package treepath provides functionality for manipulating treepaths.
+//
+// Treepaths are string-encodings of paths through trees.
+//
+// Treepaths come from
+// https://github.com/Kashomon/glift-core/blob/master/src/rules/treepath.js
+package treepath
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"unicode"
+
+	"github.com/otrego/clamshell/core/game"
+)
+
+// A Treepath is a list of variations that says how to travel through a tree of
+// moves. And has two forms a list version and astring version. First, the
+// list-version:
+//
+//    [0,1,0]
+//
+// Means we will first take the 0th variation, then we will take the 1ist
+// variation, and lastly we will take the 0th variation again. For
+// convenience, the treepath can also be specified by a string, which is where
+// the fun begins. At it's simpliest,
```

Comment:
Typo: "it's simpliest" -> "its simplest"

Reply:
Done